### ISSUE TITLE
Docker build fail

### ISSUE THREAD

**Issue by RusherRG** [Auto-DL/Auto-DL#356]
### Current Behavior

Docker build fails with the following error

```     =============================DEBUG ASSISTANCE=============================
      If you are seeing a compilation error please try the following steps to
      successfully install cryptography:
      1) Upgrade to the latest pip and try again. This will fix errors for most
         users. See: https://pip.pypa.io/en/stable/installing/#upgrading-pip
      2) Read https://cryptography.io/en/latest/installation.html for specific
         instructions for your platform.
      3) Check our frequently asked questions for more information:
         https://cryptography.io/en/latest/faq.html
      4) Ensure you have a recent Rust toolchain installed:
         https://cryptography.io/en/latest/installation.html#rust
      5) If you are experiencing issues with Rust for *this release only* you may
         set the environment variable `CRYPTOGRAPHY_DONT_BUILD_RUST=1`.
      =============================DEBUG ASSISTANCE=============================
  
  error: can't find Rust compiler
  
  If you are using an outdated pip version, it is possible a prebuilt wheel is available for this package but pip is not able to install from it. Installing from the wheel would avoid the need for a Rust compiler.
  
  To update pip, run:
  
      pip install --upgrade pip
  
  and then retry package installation.
  
  If you did intend to build this package from source, try installing a Rust compiler from your system package manager and ensure it is on the PATH during installation. Alternatively, rustup (available at https://rustup.rs) is the recommended way to download and update the Rust compiler toolchain.
  
  This package requires Rust >=1.41.0.
  ----------------------------------------
  ERROR: Failed building wheel for cryptography
```

### ⚙️ Environment

- `Python` version: 3.8.5
- `OS`: Ubuntu 20.04

### Possible Solution

The cryptography library seems to depend upon `rust`. I think it would be better if we could find an alternative to it. Maybe make use of https://github.com/pycrypto/pycrypto. And use AES instead of Fernet, although feel free to use any other.

**Comment by RusherRG** [comment]
@therajtiwari have a look at this issue

**Comment by therajtiwari** [comment]
> @therajtiwari have a look at this issue

Looks like the issue is due to the [latest version](https://cryptography.io/en/latest/changelog/#v35-0-0) of the cryptography library that rolled 2 days ago and mandates the need for a Rust compiler. We can use a specific version <3.5 which doesn't require Rust. 
Pycrypto looks like a good alternative too, can go for it.

**Comment by codewith-arijit** [comment]
> ## Bug Report
> ### Current Behavior
> 
> Docker build fails with the following error
> 
> ```
>       If you are seeing a compilation error please try the following steps to
>       successfully install cryptography:
>       1) Upgrade to the latest pip and try again. This will fix errors for most
>          users. See: https://pip.pypa.io/en/stable/installing/#upgrading-pip
>       2) Read https://cryptography.io/en/latest/installation.html for specific
>          instructions for your platform.
>       3) Check our frequently asked questions for more information:
>          https://cryptography.io/en/latest/faq.html
>       4) Ensure you have a recent Rust toolchain installed:
>          https://cryptography.io/en/latest/installation.html#rust
>       5) If you are experiencing issues with Rust for *this release only* you may
>          set the environment variable `CRYPTOGRAPHY_DONT_BUILD_RUST=1`.
>       =============================DEBUG ASSISTANCE=============================
>   
>   error: can't find Rust compiler
>   
>   If you are using an outdated pip version, it is possible a prebuilt wheel is available for this package but pip is not able to install from it. Installing from the wheel would avoid the need for a Rust compiler.
>   
>   To update pip, run:
>   
>       pip install --upgrade pip
>   
>   and then retry package installation.
>   
>   If you did intend to build this package from source, try installing a Rust compiler from your system package manager and ensure it is on the PATH during installation. Alternatively, rustup (available at https://rustup.rs) is the recommended way to download and update the Rust compiler toolchain.
>   
>   This package requires Rust >=1.41.0.
>   ----------------------------------------
>   ERROR: Failed building wheel for cryptography
> ```
> 
> ### gear Environment
> 
>     * `Python` version: 3.8.5
> 
>     * `OS`: Ubuntu 20.04
> 
> 
> ### Possible Solution
> 
> The cryptography library seems to depend upon `rust`. I think it would be better if we could find an alternative to it. Maybe make use of https://github.com/pycrypto/pycrypto. And use AES instead of Fernet, although feel free to use any other.

This problem was solved for me by changing the version (downgrading the cryptography package). Have raised a PR Kindly check.

**Comment by DasithEdirisinghe** [comment]
Hello @RusherRG I can work on this. Can you assign me. Further where can I get more resources regarding the projects

**Comment by RusherRG** [comment]
I think @codewith-arijit is working on this. I'm waiting on his reply, if no one is working on it I'll assign it to you.

**Comment by AmitGupta7580** [comment]
Hello @RusherRG I have successfully done the change of cryptography module with pycrypto module in my local repo can I create a new PR for it?

**Comment by RusherRG** [comment]
Yeah sure you can.... I will assign you this issue
